iphone/Maps/Core/TrackRecorder/TrackRecordingActivityManager.swift

```diff
@@ -22,25 +24,39 @@ final class TrackRecordingLiveActivityManager {
 
 @available(iOS 16.2, *)
 extension TrackRecordingLiveActivityManager: TrackRecordingActivityManager {
+
   func start(with info: TrackInfo) throws {
-    guard activity == nil else { return }
+    stop()
     let state = TrackRecordingLiveActivityAttributes.ContentState(trackInfo: info)
     let content = ActivityContent<TrackRecordingLiveActivityAttributes.ContentState>(state: state, staleDate: nil)
     let attributes = TrackRecordingLiveActivityAttributes()
-    activity = try LiveActivityManager.startActivity(attributes, content: content)
+    let activity = try LiveActivityManager.startActivity(attributes, content: content)
+    self.activity = activity
+    UserDefaults.standard.set(activity.id, forKey: kCurrentTrackRecordingLiveActivityIDKey)
   }
 
   func update(_ info: TrackInfo) {
-    guard let activity else { return }
+    guard let activity = activity ?? fetchCurrentActivity() else {
+      LOG(.warning, "No active TrackRecordingLiveActivity found to update.")
+      return
+    }
     let state = TrackRecordingLiveActivityAttributes.ContentState(trackInfo: info)
     let content = ActivityContent<TrackRecordingLiveActivityAttributes.ContentState>(state: state, staleDate: nil)
+    self.activity = activity
     LiveActivityManager.update(activity, content: content)
   }
 
   func stop() {
-    guard let activity else { return }
-    LiveActivityManager.stop(activity)
-    self.activity = nil
+    let activities = Activity<TrackRecordingLiveActivityAttributes>.activities
+    activities.forEach(LiveActivityManager.stop)
```

Comment:
This approach won't stop other activities that can be added later, right? E.g. navigation one.